File: Asklepios/Asklepios.Core/Entities/Departments/Room.cs
using Asklepios.Core.Entities.Patients;

namespace Asklepios.Core.Entities.Departments;

public class Room
{
    public Guid RoomId { get; set; }
    public int RoomNumber { get; set; }
    public string RoomType { get; set; }
    public int NumberOfBeds { get; set; }
    public Guid DepartmentId { get; set; }
    public Department Department { get; set; }
    public IEnumerable<Patient> Patients { get; set; }

    public Room(Guid roomId, int roomNumber, string roomType, int numberOfBeds, Guid departmentId)
    {
        RoomId = roomId;
        RoomNumber = roomNumber;
        RoomType = roomType;
        NumberOfBeds = numberOfBeds;
        DepartmentId = departmentId;
    }

    public Room()
    {
    }
}

File: Asklepios.Web/src/models/Departments/room.ts
import {PatientDto} from "@/models/Patients/patient"

export interface Room {
  roomId: string
  roomNumber: number
  roomType: string
  numberOfBeds: number
  departmentName: string
}

export class InputCreateRoom {
  roomNumber: number = 0
  roomType: string = ''
  numberOfBeds: number = 0
  departmentId: string = ''
}

export interface RoomDto {
  roomId: string
  roomNumber: number
  roomType: string
  numberOfBeds: number
  departmentId: string
}

export interface RoomListDto {
  roomId: string
  roomNumber: number
  roomType: string
  numberOfBeds: number
  departmentName: string
}

export interface RoomDetailsDto extends RoomDto {
  patients: PatientDto[];
}

export interface RoomAutocompleteDto {
  roomId: string
  roomNumber: number
  roomType: string
}

